### PR TITLE
feat: bundle typescript extension files

### PR DESCRIPTION
**Pull request by load1n9**

### CODE REVIEW

**Review by lino-levan — APPROVED** [review]

File: runtime/src/ext/console.rs [hunk 1]
```diff
@@ -23,6 +23,7 @@ impl ConsoleExt {
                 ExtensionOp::new("internal_exit", Self::internal_exit, 1),
             ],
             storage: None,
+            files: vec![include_str!("./console/mod.ts").to_string()],
```
Comment (marc2332) [inline runtime/src/ext/console.rs:26]: Unnecessary allocation here, instead of making the vec of String make it of static str.

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: core/src/runtime.rs [hunk 2]
```diff
@@ -125,6 +125,24 @@ impl<UserMacroTask> Runtime<UserMacroTask> {
...
+                                exit_with_parse_errors(diagnostics, "<runtime>", &file)
+                            }
+                        };
+                    match script_evaluation(agent, script) {
+                        Ok(_) => (),
+                        Err(_) => println!("Error in runtime"),
+                    }
+                }
+            }
+
```
Comment (marc2332) [inline core/src/runtime.rs:145]: This shouldn't be here but in `Extension::load`

File: core/src/extension.rs [hunk 3]
```diff
@@ -34,6 +34,9 @@ pub struct Extension {
     /// Storage initializer for this extension.
     /// Used to share values between the different [ExtensionOp] and multiple calls.
     pub storage: Option<ExtensionStorageInit>,
+
+    // JavaScript or Typescript files that are loaded by this extension.
+    pub files: Vec<String>,
```
Comment (marc2332) [inline core/src/extension.rs:39]:
```suggestion
    pub files: Vec<&'static str>,
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: runtime/src/ext/console.rs (hunk 1 above)
Comment (lino-levan) [reply runtime/src/ext/console.rs:26]: Making it a static str would mean that extensions would have to know what to import at compile time. Is it your intention to enforce this requirement?

Comment (load1n9) [reply runtime/src/ext/console.rs:26]: yeah

Comment (marc2332) [reply runtime/src/ext/console.rs]: > Making it a static str would mean that extensions would have to know what to import at compile time. Is it your intention to enforce this requirement?

I don't see any case where that could be needed, you may use `Cow<'static, str> to benefit from both ways instead, if you want.

**Review by marc2332 — APPROVED** [review]

**Pull request merged**